src/node.cc

```diff
@@ -584,6 +584,63 @@ const char *signo_string(int signo) {
   }
 }
 
+// These are all flags available for use with NODE_OPTIONS.
+//
+// Disallowed flags:
+//  These flags cause Node to do things other than run scripts:
+//    --version / -v
+//    --eval / -e
+//    --print / -p
+//    --check / -c
+//    --interactive / -i
+//    --prof-process
+//    --v8-options
+//  These flags are disallowed because security:
+//    --preserve-symlinks
+static const char* environment_flags[] = {
+  // Node options, sorted in `node --help` order for ease of comparison.
+  "--enable-fips",
+  "--experimental-modules",
+  "--experimenatl-repl-await",
+  "--experimental-vm-modules",
+  "--experimental-worker",
+  "--force-fips",
+  "--icu-data-dir",
+  "--inspect",
+  "--inspect-brk",
+  "--inspect-port",
+  "--loader",
+  "--napi-modules",
+  "--no-deprecation",
+  "--no-force-async-hooks-checks",
+  "--no-warnings",
+  "--openssl-config",
+  "--pending-deprecation",
+  "--redirect-warnings",
+  "--require",
+  "--throw-deprecation",
+  "--tls-cipher-list",
+  "--trace-deprecation",
+  "--trace-event-categories",
+  "--trace-event-file-pattern",
+  "--trace-events-enabled",
+  "--trace-sync-io",
+  "--trace-warnings",
+  "--track-heap-objects",
+  "--use-bundled-ca",
+  "--use-openssl-ca",
+  "--v8-pool-size",
+  "--zero-fill-buffers",
+  "-r",
+
+  // V8 options (define with '_', which allows '-' or '_')
```

Comment:
This doesn't affect what the consumers of the API receive, does it?
It looks like the consumers need to process `_` as `[\-_]` on their end, is that planned?

Reply:
Sorry; missed this.  You're right, the `Array` is just not useful for these.  I don't really see a "good" way around this using a data structure.

Instead, the API would be a function which returns `true` or `false` if a value matches an allowed `NODE_OPTIONS` environment flag.  That *may* be all we need here--Mocha certainly doesn't need to iterate over a list--but I'm curious if @benjamingr, @jdalton or @ljharb thinks that would be a sufficient API.

So, instead of 

```js
if (process.allowedEnvironmentNodeFlags.includes(arg)) {
  // do stuff
}
```

We'd have:

```js
if (process.isEnvironmentNodeFlag(arg)) {

Reply:
(I'm not opposed to changing this.  I'd rather this PR land with wider approval than *just land ASAP*.)

Reply:
I'm fine either way - although I tend to lean towards the list and guiding users to consume it the way @ChALkeR requested.

I think the a motivation for me other than API support for this API other than making libraries easier to write is the ability to determine allowed flags across node versions.

Hopefully, when we add flags tools I use would be able to pick up on them - I am not as-concerned with consumers processing `-` and `_` but I defer to @ChALkeR

Reply:
V8 flags are even tricker — there is a `--no-something` flag for every boolean `--something` flag in help.

Reply:
...which alternatively can also be spelled as `--nosomething`.